flink-yarn/src/main/java/org/apache/flink/yarn/YarnFlinkResourceManager.java

```diff
@@ -78,6 +79,9 @@
 	/** The containers where a TaskManager is starting and we are waiting for it to register */
 	private final Map<ResourceID, YarnContainerInLaunch> containersInLaunch;
 
+	/** The container where a TaskManager has been started and is running in */
+	private final Map<ResourceID, Container> containersLaunched;
```

Comment:
>You tell me, since you've authored this component. 

I think there is a misunderstanding. I did author this component but it is not mine. I know you spend some time working on this and I just want to understand your motives. I thought actually you had tried this fix since you mentioned it in the JIRA.

I think clearing the list is simply a bug and artifact of an old code design where the ResourceManager would be the central instance for TaskManager registration.